Create communities page wip

ColleenLillian opened the pull request:
#5521 

- [ ] Formatted my code with `yarn format`
- [ ] There are no warnings from `yarn lint --fix`
- [ ] There are no console warnings when running the app
- [ ] Added tests where relevant
- [ ] All tests pass
- [ ] Clicked around my changes running locally and it works
- [ ] Checked Desktop, Mobile and Tablet screen sizes

ColleenLillian committed:
Create communities page wip

ColleenLillian commented:
just making a start.
unsure about routing if you can point me in the right direction!

nabramow commented:
> just making a start. unsure about routing if you can point me in the right direction!

@ColleenLillian I believe this is where you wanna start, with the `loggedInMenu`. You should be able to do something similar to the others listed there:

app/web/components/Navigation/Navigation.tsx

ColleenLillian committed:
Add pages for communities route

ColleenLillian commented:
thanks! poked around and added an index file in pages so now it links to the new page, just need to work on making it a proper page next as it's currently basically a copy of the popup

ColleenLillian committed:
Merge branch 'develop' into cob/web/communities-page

ColleenLillian committed:
Add page title wip

nabramow commented:
Let's put the "Don't see your community..." text at the top between the header and the list of communities. That way it doesn't jump around!

<img width="915" alt="Screenshot 2025-04-23 at 8 23 11 AM" src="https://github.com/user-attachments/assets/18bdb328-6e51-41fd-9222-c54eb7e3fe52" />

nabramow commented:
There's also a mention in the ticket to add a link to the community guidelines here:

```
Link to community guidelines should go on this page: https://docs.google.com/document/d/1A9vCdKGQ_WKoyjCh5KScdGgpnighr9yL_F0-OaPT2yA/edit?usp=sharing
```

@jesseallhands where do you think this should go and what should it say?

ColleenLillian committed:
Add guidelines link

ColleenLillian commented:
I added the link for now but we can decide what the text should be.
Do we just want to link to the community builder toolkit or also to the help centre community guidelines?
I don't think the flow is super clear right now. I guess we want people to read the toolkit and guidelines first before starting their own community/becoming a community builder?

nabramow commented on app/web/features/communities/CommunitiesPage/CommunitiesPage.tsx:
```diff
@@ -0,0 +1,280 @@
+const COMMUNITY_GUIDELINES_LINK =
+  "https://docs.google.com/document/d/1A9vCdKGQ_WKoyjCh5KScdGgpnighr9yL_F0-OaPT2yA/edit?usp=sharing";
+
+const useStyles = makeStyles((theme) => ({
+  root: {
+    display: "flex",
+    flexDirection: "row",
+    alignItems: "flex-start",
+    "& > * + *": {
+      marginInlineStart: theme.spacing(2),
+    },
+  },
+  list: {
+    minWidth: "10rem",
+  },
+  loader: {
+    margin: theme.spacing("auto", 2),
+  },
+  selected: {
+    fontWeight: "bold",
+  },
```
Would you mind switching the styles for this to the new `styled` format? We're working on getting rid of `makeStyles` so we can upgrade the package so try to do new components the new way.

If you search `styled` or `Styled` you should see some examples! Just shout if you have questions about it I've done a billionnnnn conversions.

nabramow commented on app/web/features/dashboard/Hero/HeroLinks.tsx:
```diff
@@ -83,21 +83,29 @@ export default function HeroLinks() {
+      {/*<CommunitiesDialog
         isOpen={communitiesDialogOpen}
         onClose={() => setCommunitiesDialogOpen(false)}
-      />
```
Don't forget to remove this!

nabramow commented on app/web/features/dashboard/Hero/HeroLinks.tsx:
```diff
@@ -83,21 +83,29 @@ export default function HeroLinks() {
             </StyledLink>
 
-            <MuiLink
+            {/*<MuiLink
               underline="none"
               component="button"
               className={classes.tabAppearance}
               onClick={() => setCommunitiesDialogOpen(true)}
             >
               {t("browse_communities")}
```
Don't forget to remove any commented out things!

ColleenLillian replied:
yep! as long as everything works as it should i'll go back through and remove stuff!

nabramow commented on app/web/features/dashboard/Hero/HeroLinks.tsx:
```diff
@@ -83,21 +83,29 @@ export default function HeroLinks() {
             >
               {t("browse_communities")}
-            </MuiLink>
+            </MuiLink>*/}
```
Aaaand this

nabramow commented on app/web/features/dashboard/CommunitiesSection.tsx:
```diff
@@ -21,7 +22,7 @@ const useStyles = makeStyles((theme) => ({
   const { t } = useTranslation([GLOBAL, DASHBOARD]);
   const classes = useStyles();
-  const [isCommunitiesDialogOpen, setIsCommunitiesDialogOpen] = useState(false);
+  //const [isCommunitiesDialogOpen, setIsCommunitiesDialogOpen] = useState(false);
```
Don't forget to remove this

nabramow commented on app/web/features/communities/CommunitiesPage/CommunitiesPage.tsx:
```diff
@@ -0,0 +1,280 @@
+        </Trans>
+      </Typography>
+
+      <div className={classes.root}>
+        {cachedQueryResults.map((query, index) => (
+          <BrowserColumn
+            key={index}
+            parent={selected?.[index - 1] ?? globalCommunityQuery.data}
+            communities={query.data}
+            handleClick={(community) => handleClick(community, index)}
+            selected={selected[index]?.communityId}
+          />
+        ))}
+        {query.isLoading ? ( // div prevents overflow scrollbar from spinner
+          <div className={classes.loader}>
+            <CenteredSpinner />
+          </div>
+        ) : query.isSuccess && globalCommunityQuery.isSuccess ? (
+          <div ref={lastColumnRef}>
+            <BrowserColumn
+              parent={
+                selected?.[selected.length - 1] ?? globalCommunityQuery.data
+              }
+              communities={query.data.pages.flatMap(
+                (page) => page.communitiesList,
+              )}
```
Would you mind putting the communities browser in its own component? You can call it `CommunitiesBrowser`, then import that component in here.

Separates the logic more nicely that way I think.

nabramow commented on app/web/features/communities/CommunitiesPage/CommunitiesPage.tsx:
```diff
@@ -0,0 +1,280 @@
+  >([]);
+
+  const lastColumnRef = useRef<HTMLDivElement>(null);
+  useEffect(() => {
+    setTimeout(
+      () =>
+        lastColumnRef.current?.scrollIntoView({
+          behavior: "smooth",
+          inline: "end",
+        }),
+      50,
+    );
+  }, [selected]);
```
When I first load the Communities page it scrolls me down below the heading which is a bit confusing. I think because of this?

When you separate out the CommunityBrowser maybe this logic can go in there so it only scrolls within a container.

nabramow commented:
In Chrome when I turn on responsive mobile view and then choose Asia - Pacific => Australia it scrolls out weird like this:

<img width="234" alt="Screenshot 2025-04-24 at 9 08 51 PM" src="https://github.com/user-attachments/assets/08c36e55-1fca-41f8-89f1-b060f7d8c7ee" />

I assume this will happen on Desktop on prod too where we have more communities and sub communities.

I think what you want to do here is wrap this in two divs. An outerwrapper with css `overflow: "hidden"` and then the CommunityBrowser should have `overflow: "auto"` so the scrolling happens within a container rather than stretching our the whole page.

ColleenLillian committed:
Merge branch 'develop' into cob/web/communities-page

ColleenLillian committed:
Separate community browser and use styled

ColleenLillian committed:
Format

ColleenLillian commented:
Okkk I cleaned it up and separated the logic. I still need to work out why it's scrolling weird and fix that!
It's mostly on mobile that it doesn't start at the top of the page on load, and then scrolls to the bottom when you select a sub community.

nabramow commented:
> Okkk I cleaned it up and separated the logic. I still need to work out why it's scrolling weird and fix that! It's mostly on mobile that it doesn't start at the top of the page on load, and then scrolls to the bottom when you select a sub community.

I think it's okay if it scrolls to the sub-community after selecting one. The issue was more that on initial page load it scrolls, which is kinda of confusing, and it messed up the formatting on mobile.

Might be a good way to go about it to first remove the scrolling totally. Then add it back in to the new separate community browser component (if it seems needed). That might be adding `scrollIntoView` onSelect instead of in the useEffect, for example.

ColleenLillian committed:
Merge branch 'develop' into cob/web/communities-page

ColleenLillian committed:
Remove scroll and update text

ColleenLillian commented:
I updated the text and links to match Jesse's suggestions (but I put it in separate sections so it looks nicer on mobile), and I just removed the scroll. lmk what else we could improve!

nabramow commented:
Okay we're getting closer! A few things:

* In `app/web/pages/communities/index.tsx` can you add `NOTIFICATIONS` to the array in `serverSideTranslations`? That will stop the missing translation warnings in the console. (Just happening because we merged notifications feed while this was in progress and didn't have that page there)

* The line spaces between the paragraphs is looking different to me than what Jesse sent me (unless he sent you something different haha).

This is what I see:

<img width="1191" alt="Screenshot 2025-05-02 at 10 00 13 PM" src="https://github.com/user-attachments/assets/f497b3d1-e8e2-404b-b969-ca3aed792923" />

This is what Jesse sent me:
![image (4)](https://github.com/user-attachments/assets/7be5985c-d2ac-47c1-8735-16e3d904c5b0)

nabramow commented:
It looks like the `CommunityBrowser` now has the next view opening to the bottom instead of to the right. This makes it look like it has a double:

<img width="397" alt="Screenshot 2025-05-02 at 10 01 48 PM" src="https://github.com/user-attachments/assets/a6fd4713-1403-48c2-93b9-238696213c9c" />

We want it to stay to the right like it was before. If it expands the screen size, it should scroll just the CommunityBrowser, not the whole page.

You got the scrolling working great though 🎉, we just want it horizontal instead of vertical. I think you probably just need to add `display: flex` to the `InnerWrapper` and `width: 100%` to `OuterWrapper`.

I'll do a last code review once we get all the UI sorted, but from first glance it's looking okay!

ColleenLillian committed:
Fix layout and add translation

ColleenLillian commented:
chatted with Jesse and agreed to leave a bit of separation to make the links clearer and avoid it looking too text-heavy. hopefully the scroll and translations are fixed now!

nabramow commented on app/web/features/communities/CommunitiesPage/CommunitiesPage.tsx:
```diff
@@ -0,0 +1,88 @@
+}));
+
+const CommunitiesPage = () => {
+  const { t } = useTranslation([COMMUNITIES, DASHBOARD, GLOBAL]);
```
I don't think you're using `GLOBAL` here (i.e. translations that start with "global:"`? Double check me but you can remove if not!

ColleenLillian replied:
yeah i think you're right. I'll check over the translations again since a few parts were just what I copied from the original component

nabramow commented on app/web/features/dashboard/CommunityBrowser.tsx:
```diff
@@ -10,36 +10,51 @@ import {
-    "& > * + *": {
-      marginInlineStart: theme.spacing(2),
-    },
+const OuterWrapper = styled("div")(({ theme }) => ({
+  display: "flex",
+  flexDirection: "row",
+  alignItems: "flex-start",
+  width: "100%",
+  "& > * + *": {
+    marginInlineStart: theme.spacing(2),
```
Thank you so much for converting this to the new format! Saves me so much time later!

nabramow commented on app/web/features/dashboard/CommunityBrowser.tsx:
```diff
@@ -99,45 +102,49 @@ export default function CommunityBrowser() {
-      )}
-    </div>
+        ))}
+        {query.isLoading ? ( // div prevents overflow scrollbar from spinner
```
What does this comment mean? I don't totally understand.

ColleenLillian replied:
It's just something that was already there when I copied it over from the popup, so I'm not sure either!

nabramow commented on app/web/resources/locales/en.json:
```diff
@@ -125,6 +125,7 @@
     "badges": "Badges",
     "blog": "Blog",
     "community": "Community",
+    "communities": "Communities",
```
I think we have three "communities" strings now, haha. Can we remove this one and use "communities_title" from the communities en.json instead?

Why does this matter? For the other languages, in general good to use existing strings we already have, otherwise we create more that only has an English translation (since people manually translate the strings to other languages in Weblate).

ColleenLillian replied:
haha yes ok, I got so lost trying to find all the translations so definitely need to clean that up

nabramow commented:
Okay this is looking great! Only really minor formatting things this go. Looks like you also have some failing tests, but once that's addressed I'm happy to approve! Let me know if you need help with the test stuff.

ColleenLillian committed:
Merge branch 'develop' into cob/web/communities-page

ColleenLillian committed:
Fix translations

ColleenLillian commented:
had a bit of faff with changing the navbar translation, so I just removed the other one instead fyi (so now I do need global after all ha)

ColleenLillian committed:
Fix failing test

nabramow commented:
> had a bit of faff with changing the navbar translation, so I just removed the other one instead fyi (so now I do need global after all ha)

Ahh so every time you add a new translation, you need to restart yarn for it to show as it's a wrapper around the app. So just stop the frontend from running, then do `yarn start` again and any new translations or changed ones should show!

ColleenLillian commented:
> > had a bit of faff with changing the navbar translation, so I just removed the other one instead fyi (so now I do need global after all ha)
> 
> Ahh so every time you add a new translation, you need to restart yarn for it to show as it's a wrapper around the app. So just stop the frontend from running, then do `yarn start` again and any new translations or changed ones should show!

yeah thanks i did that! it was just that I had them listed in the wrong order in the end

nabramow commented:
Okay I'm really sorry, I found a few more things when comparing with prod. 

* It looks like we lost the blue link color for the "header" community somewhere? 
* The divider grey lines also have a little space between them instead of running together. I think it makes it look cleaner.
* The one you selected is bold on prod.

This is how it looks on prod:

<img width="591" alt="Screenshot 2025-05-05 at 8 10 17 AM" src="https://github.com/user-attachments/assets/302e9889-a1db-4a8c-82fd-a2e6bef40f3a" />

ColleenLillian committed:
Fix styles

nabramow commented on app/web/features/dashboard/CommunityBrowser.tsx:
```diff
@@ -153,61 +171,56 @@ function BrowserColumn({
+            >
+              <StyledListItemText
+                primary={parent.name}
+                sx={{
+                  color: (theme) => theme.palette.primary.main,
+                }}
```
```suggestion
                sx={{
                  color:  theme.palette.primary.main,
                }}
```

and add this import up top

`import { theme } from "theme";
`

ColleenLillian committed:
Merge branch 'develop' into cob/web/communities-page

ColleenLillian committed:
Import theme

nabramow reviewed (approved):
Okay this looks good to go, well done! It will be so much easier for users to find the communities pages now!
